specification/codesigning/CodeSigning/main.tsp

```diff
@@ -0,0 +1,221 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-autorest";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Autorest;
+
+@useAuth(
+  OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+      scopes: ["https://codesigning.azure.net/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Azure.CodeSigning service",
+})
+@versioned(Azure.CodeSigning.Versions)
+@server(
+  "https://{region}.codesigning.azure.net/",
+  "Azure CodeSigning is a service that provides managed code signing for all.",
+  {
+    @doc("The region where we will sign code.")
+    region: string,
+  }
+)
+@doc("Azure CodeSigning is a service that provides managed code signing for all.")
+namespace Azure.CodeSigning;
+
+enum Versions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  v2022_06_15_preview: "2022-06-15-preview",
+}
+
+interface CertificateProfileOperations {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is a custom operation status endpoint."
+  @summary("Gets the status of a sign operation.")
+  @doc("Gets the status of a sign operation.")
+  @example(
+    "./examples/CertificateProfile_GetSignOperation.json",
+    "GetSignOperation"
+  )
+  @route("/codesigningaccounts/{codeSigningAccountName}/certificateprofiles/{certificateProfileName}/sign/{operationId}")
+  getSignOperation is Foundations.GetOperationStatus<
+    CodeSignStatus,
+    OperationStatus
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is a custom operation status endpoint."
```

Comment:
The explanation for this suppression looks wrong -- looks like it was just copied from above. Please consider using an operation template or provide a better explanation here.

If you do not change to use an operation template, you must add an api-version parameter and a default (error) response.

Reply:
This will be resolved using the suggestion below.